streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedCogroupedKStreamImpl.java

```diff
@@ -0,0 +1,182 @@
+
+    TimeWindowedCogroupedKStreamImpl(final Windows<W> windows,
+                                     final InternalStreamsBuilder builder,
+                                     final Set<String> sourceNodes,
+                                     final String name,
+                                     final Serde<K> keySerde,
+                                     final CogroupedStreamAggregateBuilder<K, V> aggregateBuilder,
+                                     final StreamsGraphNode streamsGraphNode,
+                                     final Map<KGroupedStreamImpl<K, ?>, Aggregator<? super K, ? super Object, V>> groupPatterns) {
+        super(name, null, null, sourceNodes, streamsGraphNode, builder);
+        this.windows = windows;
+        this.aggregateBuilder = aggregateBuilder;
+        this.groupPatterns = groupPatterns;
+    }
+
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer) {
+        return aggregate(initializer, Materialized.with(keySerde, null));
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Materialized<K, V, WindowStore<Bytes, byte[]>> materialized) {
+        return aggregate(initializer, NamedInternal.empty(), materialized);
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Named named) {
+        return aggregate(initializer, named, Materialized.with(keySerde, null));
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Named named,
+                                            final Materialized<K, V, WindowStore<Bytes, byte[]>> materialized) {
+        Objects.requireNonNull(initializer, "initializer can't be null");
+        Objects.requireNonNull(named, "named can't be null");
+        Objects.requireNonNull(materialized, "materialized can't be null");
+        final MaterializedInternal<K, V, WindowStore<Bytes, byte[]>> materializedInternal = new MaterializedInternal<>(
+            materialized,
+            builder,
+            AGGREGATE_NAME);
+        return aggregateBuilder.build(
+            groupPatterns,
+            initializer,
+            NamedInternal.empty(),
+            materialize(materializedInternal),
+            materializedInternal.keySerde() != null ?
+                    new FullTimeWindowedSerde<>(materializedInternal.keySerde(), windows.size())
+                    : null,
+            materializedInternal.valueSerde(),
+            materializedInternal.queryableStoreName(),
+            windows,
+            null,
+            null);
+    }
+
+    @SuppressWarnings("deprecation")
+    // continuing to support Windows#maintainMs/segmentInterval in fallback mode
+    private StoreBuilder<TimestampedWindowStore<K, V>> materialize(
```

Comment:
This method seems to be the exact same as `TimeWindowedKStreamImpl#materialize()` -- we should share the code.

Reply:
It almost is. But there are a few Type differences that make it so I can't use it directly and if I were to adapt `TimeWindowedKStreamImpl#materialize()` it would become less type safe